pkg/sql/parser/eval.go

```diff
@@ -2088,6 +2088,11 @@ func (expr *FuncExpr) Eval(ctx *EvalContext) (Datum, error) {
 
 	res, err := expr.fn.fn(ctx, args)
 	if err != nil {
+		// If we are facing a retry error, propagate it unchanged, so that
+		// the executor can see it with the right type.
+		if _, ok := err.(*roachpb.RetryableTxnError); ok {
```

Comment:
I have tried to switch to `Wrapf` and it becomes a rather non-trivial patch: because the sql stuff is run as a closure by `client.Txn`, we need to plumb `errors.Cause()` everywhere in the `client` package where there is a check for `RetryableTxnError`. Is this what we want?
Or should I add a new predicate `IsRetryableTxnError()` in `roachpb` that calls `Cause` and tests the cast, and use that everywhere?